### PR TITLE
CLS data

### LINKED ISSUE #70: CLS benchmark data model
#69 

Updates to the data model to store benchmark data
```sql
CREATE TABLE cls_benchmark_data (
  cls_benchmark_data_id SERIAL NOT NULL UNIQUE,
  frontline_survey_question_id INT NOT NULL UNIQUE,
  mean_score_question INT,
  mean_score_answer_1 INT,
  mean_score_answer_2 INT,
  mean_score_answer_3 INT,
  mean_score_answer_4 INT,
  mean_score_answer_5 INT,
  mean_score_answer_6 INT,

  CONSTRAINT cls_benchmark_data_pk PRIMARY KEY (cls_benchmark_data_id),
  CONSTRAINT cls_benchmark_data_to_frontline_survery_question_fk FOREIGN KEY (frontline_survey_question_id) REFERENCES frontline_survey_question ON DELETE CASCADE
)
```

### LINKED ISSUE #71: CLS benchmark ingestion
#69 

Need to write a script that will stick the data into the DB. Data will be delivered via e-mail, possibly unstructured.

### LINKED ISSUE #72: CLS benchmark endpoint
#69 

Need to expose the CLS benchmark data on the API, via an endpoint that accepts a pre-authenticated token (since Frontline will be the only ones using it).

### PR DESCRIPTION
**Pull request by eliascodes**
### Fixes
Fixes #70 
Fixes #71 
Fixes #72 

### Changes
* Add benchmark tables to DB and benchmark data to seeds
* Add endpoint to grab benchmark data

### CODE REVIEW

File: database/migrations/sql/20180805_14_create_cls_survey_tables.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Benchmark survey data tables
+ * National and regional benchmarks for survey questions
+ */
+CREATE TABLE cls_survey_benchmark_data (
+  cls_survey_benchmark_data_id SERIAL NOT NULL UNIQUE,
+  frontline_survey_question_id INT NOT NULL,
+  community_business_region_id INT NOT NULL,
+  score                        FLOAT NOT NULL,
```
Comment (astroash) [inline database/migrations/sql/20180805_14_create_cls_survey_tables.sql]: can we add year?

File: database/migrations/sql/20180805_14_create_cls_survey_tables.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  CONSTRAINT cls_survey_benchmark_data_community_business_region_fk FOREIGN KEY (community_business_region_id) REFERENCES community_business_region ON DELETE CASCADE,
+  CONSTRAINT cls_benchmark_data_unique_per_region_per_question      UNIQUE (frontline_survey_question_id, community_business_region_id)
+);
+
+CREATE TABLE nps_survey_benchmark_data (
+  nps_survey_benchmark_data_id SERIAL NOT NULL UNIQUE,
+  frontline_survey_question_id INT NOT NULL,
+  detractors                   FLOAT NOT NULL,
+  passives                     FLOAT NOT NULL,
+  promoters                    FLOAT NOT NULL,
```
Comment (astroash) [inline database/migrations/sql/20180805_14_create_cls_survey_tables.sql]: can we add year?

File: database/seeds/testing/21_frontline_survey_questions.seed.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+exports.seed = (knex) =>
```
Comment (astroash) [inline database/seeds/testing/21_frontline_survey_questions.seed.js]: can this file name have `test` prepended to make it clear this is test data 

eg `21_test_frontline.....`

File: src/models/community_business.ts [hunk 4]
```diff
@@ -225,6 +234,14 @@ export const CommunityBusinesses: CommunityBusinessCollection = {
       .where(preProcessCb(client)(o));
   },
 
+  async fromOrganisation (client: Knex, o: Organisation) {
+    const [{ community_business_id: id }] = await client('community_business')
```
Comment (astroash) [inline src/models/community_business.ts]: community_business_id no longer exists in my PR.

File: src/models/community_business.ts [hunk 5]
```diff
@@ -225,6 +234,14 @@ export const CommunityBusinesses: CommunityBusinessCollection = {
       .where(preProcessCb(client)(o));
   },
 
+  async fromOrganisation (client: Knex, o: Organisation) {
```
Comment (astroash) [inline src/models/community_business.ts]: This method is not needed as I've added organisation_id to the `CommunityBusinesses` model.

File: database/seeds/testing/21_frontline_survey_questions.seed.js (hunk 3 above)
Comment (eliascodes) [reply database/seeds/testing/21_frontline_survey_questions.seed.js]: I will push the real survey questions soon, so this will be unnecessary

**Comment by eliascodes** [comment]
@astroash this is ready to go, not bothered about <1% coverage hits

File: database/migrations/sql/20180805_14_create_cls_survey_tables.sql (hunk 1 above)
Comment (astroash) [reply database/migrations/sql/20180805_14_create_cls_survey_tables.sql]: @eliascodes

**Pull request merged**